src/sysusers/sysusers.c

```diff
@@ -924,6 +991,12 @@ static int write_files(Context *c) {
 
         assert(c);
 
+        /* Immediately mark users and groups as not failed, iff there is nothing to write. */
+        if (ordered_hashmap_isempty(c->todo_uids))
+                *ret_users_failed = false;
```

Comment:
i think it would be nicer to initialize the users_failed parameter unconditionally here, not just if empty. i.e. don't expect the param to be initialized on invocation at all.